CI failing

<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release are affected -->

# Summary
CI failing with installing docker compose. 
<!--- Explain the problem briefly below -->

# Issue Type

- Bug Report

# Ansible, Collection, Private Automation Hub details
CI

# Desired Behavior

<!--- Describe what you expected to happen when running the steps above -->

# Actual Behavior

<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->


```console (error)
Run curl -SL https://github.com/docker/compose/releases/download/v2.20.3/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose && sudo chmod +x /usr/local/bin/docker-compose
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Warning: Failed to create the file /usr/local/bin/docker-compose: Permission 
Warning: denied

  0 56.6M    0 163[7](https://github.com/redhat-cop/eda_configuration/actions/runs/8075852299/job/22063369246#step:6:8)5    0     0  70750      0  0:13:5[9](https://github.com/redhat-cop/eda_configuration/actions/runs/8075852299/job/22063369246#step:6:10) --:--:--  0:[13](https://github.com/redhat-cop/eda_configuration/actions/runs/8075852299/job/22063369246#step:6:14):59 70750
curl: (23) Failure writing output to destination
Error: Process completed with exit code 23.
```

--- a/.github/workflows/ci_standalone_versioned.yml
+++ b/.github/workflows/ci_standalone_versioned.yml
@@ -38,7 +38,7 @@ jobs:
         run: sudo apt -y update
 
       - name: Install docker-compose
-        run: curl -SL https://github.com/docker/compose/releases/download/v2.20.3/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose && sudo chmod +x /usr/local/bin/docker-compose
+        run: sudo curl -SL https://github.com/docker/compose/releases/download/v2.20.3/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose && sudo chmod +x /usr/local/bin/docker-compose
 
       - name: collect system info
         run: whoami; id; pwd; ls -al; uname -a ; df -h .; mount ; cat /etc/issue; docker --version ; ps aux | fgrep -i docker; ls -al /var/run/containerd/containerd.sock
@@ -53,7 +53,12 @@ jobs:
 
       - name: Start stack
         working-directory: eda-server/tools/docker
-        run: docker compose -p eda -f docker-compose-stage.yaml up -d --wait
+        run: |
+          docker-compose -p eda -f docker-compose-stage.yaml up -d
+          while ! curl -s http://localhost:8000/_healthz | grep -q "OK"; do
+            echo "Waiting for API to be ready..."
+            sleep 1
+          done
 
       - name: Move ansible.cfg to root
         run: mv .github/files/ansible.cfg .

--- a/changelogs/fragments/bugfix.yml
+++ b/changelogs/fragments/bugfix.yml
@@ -1,4 +1,5 @@
 ---
 bugfixes:
   - fixed a bug where resolve_name_to_id data was not defined
+  - fixed a bug with the API returning multiple items because it only matches the name starting with the value
 ...

--- a/changelogs/fragments/user_token.yml
+++ b/changelogs/fragments/user_token.yml
@@ -1,5 +1,5 @@
 ---
-    major_changes:
-      - Added user_token module
-      - Added user_token role
-    ...
+major_changes:
+  - Added user_token module
+  - Added user_token role
+...

--- a/plugins/module_utils/eda_module.py
+++ b/plugins/module_utils/eda_module.py
@@ -302,14 +302,23 @@ class EDAModule(AnsibleModule):
                 self.fail_wanted_one(response, endpoint, new_kwargs.get("data"))
         elif response["json"]["count"] > 1:
             if name_or_id:
-                # Since we did a name or ID search and got > 1 return something if the id matches
+                # Since we did a name or ID search and got > 1 return something if the id matches or the name matches exactly
+                exact_matches = []
                 for asset in response["json"]["results"]:
                     if str(asset["id"]) == name_or_id:
                         return self.existing_item_add_url(asset, endpoint, key=key)
+                    if str(asset[name_field]) == name_or_id:
+                        exact_matches.append(self.existing_item_add_url(asset, endpoint, key=key))
+                # If there is one exact name match then return that
+                if len(exact_matches) == 1:
+                    return exact_matches[0]
 
-            # We got > 1 and either didn't find something by ID (which means multiple names)
+            # We got > 1 and either didn't find something by ID or exact name (which means multiple names)
             # Or we weren't running with a or search and just got back too many to begin with.
-            self.fail_wanted_one(response, endpoint, new_kwargs.get("data"))
+            if allow_none:
+                return None
+            else:
+                self.fail_wanted_one(response, endpoint, new_kwargs.get("data"))
 
         return self.existing_item_add_url(response["json"]["results"][0], endpoint, key=key)
 
@@ -763,7 +772,7 @@ class EDAModule(AnsibleModule):
         return self.get_one(endpoint, name_or_id=name_or_id, allow_none=False, **kwargs)
 
     def resolve_name_to_id(self, endpoint, name_or_id, data=None):
-        return self.get_exactly_one(endpoint, name_or_id, **{"data": data})["id"]
+        return self.get_exactly_one(endpoint, name_or_id, **{"data": data if data else {}})["id"]
 
     def objects_could_be_different(self, old, new, field_set=None, warning=False):
         if field_set is None:

--- a/plugins/modules/rulebook_activation.py
+++ b/plugins/modules/rulebook_activation.py
@@ -124,7 +124,7 @@ def main():
         extra_vars=dict(type="dict"),
         enabled=dict(type="bool", default="true"),
         state=dict(choices=["present", "absent", "restarted"], default="present"),
-        awx_token=dict(no_log=True),
+        awx_token=dict(no_log=False),
     )
 
     # Create a module for ourselves

--- a/roles/project_sync/tasks/main.yml
+++ b/roles/project_sync/tasks/main.yml
@@ -24,7 +24,7 @@
   vars:
     ansible_async_dir: '{{ eda_configuration_async_dir }}'
 
-- name: "Create project | Wait for finish the project creation"
+- name: "Sync project | Wait for finish syncing the project creation"
   ansible.builtin.async_status:
     jid: "{{ __projects_sync_job_async_result_item.ansible_job_id }}"
   register: __projects_sync_job_async_result

--- a/tests/playbooks/eda_configs/eda_auth.yml
+++ b/tests/playbooks/eda_configs/eda_auth.yml
@@ -1,6 +1,6 @@
 #  User may add tower auth creds to this file and encrypt it using `ansible-vault`
 ---
-eda_hostname: http://localhost:8080/
+eda_hostname: http://localhost:8000
 eda_username: admin
 eda_password: testpass
 eda_validate_certs: false

--- a/tests/playbooks/eda_configs/eda_rulebook_activations.yml
+++ b/tests/playbooks/eda_configs/eda_rulebook_activations.yml
@@ -4,20 +4,11 @@ eda_rulebook_activations:
     description: Hook to listen for changes in GitHub
     project: EDA_example
     rulebook: git-hook-deploy-rules.yml
+    awx_token: my_user_token
     decision_environment: Automation Hub Default Decision Environment
     extra_vars:
       provider: github-local
       repo_url: https://github.com/ansible/ansible-rulebook.git
-    enabled: false
+    enabled: true
     state: present
-  - name: Github Hook
-    description: Hook to listen for changes in GitHub
-    project: EDA_example
-    rulebook: git-hook-deploy-rules.yml
-    decision_environment: Automation Hub Default Decision Environment
-    extra_vars:
-      provider: github-local
-      repo_url: https://github.com/ansible/ansible-rulebook.git
-    enabled: false
-    state: restarted
 ...

--- a/tests/playbooks/testing_collections_playbook.yml
+++ b/tests/playbooks/testing_collections_playbook.yml
@@ -17,7 +17,7 @@
     - name: Verify EDA Controller is up
       ansible.builtin.uri:
         validate_certs: false
-        url: "{{ eda_hostname }}/eda"
+        url: "{{ eda_hostname }}/api/eda/v1/users/me/"
         url_username: "{{ eda_username }}"
         url_password: "{{ eda_password }}"
         status_code:

--- a/tests/playbooks/testing_collections_playbook.yml
+++ b/tests/playbooks/testing_collections_playbook.yml
@@ -27,24 +27,57 @@
     - name: Create project
       ansible.builtin.include_role:
         name: project
+        apply:
+          tags:
+            - project
+      tags:
+        - project
 
     - name: Sync project
       ansible.builtin.include_role:
         name: project_sync
+        apply:
+          tags:
+            - project
+            - project_sync
+      tags:
+        - project
+        - project_sync
 
     - name: Create credential
       ansible.builtin.include_role:
         name: credential
+        apply:
+          tags:
+            - credential
+      tags:
+        - credential
 
     - name: Create decision environment
       ansible.builtin.include_role:
         name: decision_environment
-
-    - name: Create rulebook activations
-      ansible.builtin.include_role:
-        name: decision_environment
+        apply:
+          tags:
+            - decision_environment
+      tags:
+        - decision_environment
 
     - name: Create user tokens
       ansible.builtin.include_role:
         name: user_token
+        apply:
+          tags:
+            - user_token
+      tags:
+        - user_token
+
+    - name: Create rulebook activations
+      ansible.builtin.include_role:
+        name: rulebook_activation
+        apply:
+          tags:
+            - rulebook_activation
+      tags:
+        - rulebook_activation
+
 ...

